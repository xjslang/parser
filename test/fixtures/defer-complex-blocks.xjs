function complexDeferScenario() {
  console.log('start')
  
  // Defer en diferentes tipos de bloques
  for (let i = 0; i < 2; i++) {
    defer {
      console.log('for defer:', i)
    }
  }
  
  if (true) {
    defer {
      console.log('if defer')
    }
  }
  
  try {
    defer {
      console.log('try defer')
    }
    throw new Error('test')
  } catch (e) {
    defer {
      console.log('catch defer')
    }
  }
  
  console.log('end')
}

complexDeferScenario()
