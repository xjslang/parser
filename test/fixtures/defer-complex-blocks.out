start
end
catch defer
try defer
if defer
for defer: 1
for defer: 0
